mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h

```diff
@@ -20,6 +20,8 @@
 
 namespace mlir {
 
+enum class DataType { f32, f16, bf16 };
```

Comment:
I feel like it make better sense if you use the existing type infrastructure to handle this: see `mlir/lib/IR/Types.cpp`

That should give you a free implementation to get bitwidth and with which you can decide vectorization size based on bitwidth.

Reply:
@jerryyin is this `ConvContext` used by external clients? if not, I'll switch to MLIR `Type`.

Reply:
It's hidden behind and is not exposed in the API. It's up to us how this is implemented.

Reply:
Got it. Let me revise the implementation, using `mlir::Type`.